File: webtester/java/src/test/java/com/mytesting/pages/FilterPanel.java

package com.mytesting.pages;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.How;
import org.openqa.selenium.support.PageFactory;
import org.openqa.selenium.support.ui.WebDriverWait;


public class FilterPanel extends BasePage {
 
    @FindBy(how=How.CSS, using="input[name='min_price']")
    private WebElement minPriceInput; 

	@FindBy(how=How.CSS, using="input[name='max_price']")
	private WebElement maxPriceInput; 
	
    @FindBy(how=How.CSS, using="input[value='price_asc']")
    private WebElement sortByPriceAscInputRadio; 
    
    @FindBy(how=How.CSS, using="input[value='price_desc']")
    private WebElement sortByPriceDescInputRadio; 
    
	public FilterPanel(WebDriver driver) {
		super(driver);
	}
	
	public void setMaxPrice(String maxPrice) {
		
		advancedActionBot.waitToBeEnabledAndEnterText(maxPriceInput, maxPrice);
	}
	
	public void setMinPrice(String minPrice) {
		
		advancedActionBot.waitToBeEnabledAndEnterText(minPriceInput, minPrice);
	}
	
	public void clickSortByPriceAscInputRadio() {
		
		advancedActionBot.waitToBeEnabledAndClick(sortByPriceAscInputRadio);
	}
	
	public void clickSortByPriceDescInputRadio() {
		
		advancedActionBot.waitToBeEnabledAndClick(sortByPriceDescInputRadio);
	}
	
	


}

File: webtester/nodejs/features/page_objects/filter_panel.js

const By = require('selenium-webdriver').By;
const until = require('selenium-webdriver').until;

var filterPanel = function () {

    const elements = {
        minPriceInput: By.css('input[id^="minprice-Price"]'),
        maxPriceInput: By.css('input[id^="maxprice-Price"]'),
        sortByPriceAscInputRadio: By.xpath('//div[@class="price_asc"]/input'),
        sortByPriceDescInputRadio: By.xpath('//div[@class="price_desc"]/input')
    };

    filterPanel.prototype.setMinPrice = function (context, minPrice) {
        context.driver.wait(until.elementLocated(elements.minPriceInput));
        return context.driver.findElement(elements.minPriceInput).sendKeys(minPrice);
    };

    filterPanel.prototype.setMaxPrice = function (context, maxPrice) {
        context.driver.wait(until.elementLocated(elements.maxPriceInput));
        return context.driver.findElement(elements.maxPriceInput).sendKeys(maxPrice);
    };

    filterPanel.prototype.clickSortByPriceAscInputRadio = function (context) {
        context.driver.wait(until.elementLocated(elements.sortByPriceAscInputRadio));
        return context.driver.findElement(elements.sortByPriceAscInputRadio).click();
    };

    filterPanel.prototype.clickSortByPriceDescInputRadio = function (context) {
        context.driver.wait(until.elementLocated(elements.sortByPriceDescInputRadio));
        return context.driver.findElement(elements.sortByPriceDescInputRadio).click();
    };



};

module.exports = new filterPanel();